### PR TITLE
Improve code coverage

### PR DESCRIPTION
**Pull request by martincostello**
* Add unit tests for uncovered members and code branches.
  * Remove code that was only exercised by tests and benchmarks and wasn't otherwise used.
  * If the bucket name is null for some reason, don't bother trying to send a stat.
  * Rename file to match type.

### CODE REVIEW

File: src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs [hunk 1]
```diff
@@ -43,7 +43,7 @@ public void Timing(long duration, double sampleRate, string bucket)
 
         private void SendMessage(double sampleRate, in StatsDMessage msg)
         {
-            bool shouldSendMessage = sampleRate >= DefaultSampleRate || sampleRate > Random.NextDouble();
+            bool shouldSendMessage = (sampleRate >= DefaultSampleRate || sampleRate > Random.NextDouble()) && msg.StatBucket != null;
```
Comment (AnthonySteele) [inline src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs]: Is this a place to use `! String.IsNullOrEmpty(msg.StatBucket)` ?

**Comment by AnthonySteele** [comment]
Typically "Endpoint" not "EndPoint" : https://en.wikipedia.org/wiki/Endpoint

**Comment by AnthonySteele** [comment]
Are there any changes that could be considered "breaking" here? Would like to avoid those now.

File: src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs (hunk 1 above)
Comment (martincostello) [reply src/JustEat.StatsD/Buffered/BufferBasedStatsDPublisher.cs]: If we want to. I was using null to try and force the exception branch of the code but couldn’t, it was just pointlessly sending a metric, so just added this in instead.

**Comment by martincostello** [comment]
There shouldn’t be any breaking changes here. All I did is delete an internal class and make something not send bad data.

**Pull request merged**